networking_odl/common/utils.py

```diff
@@ -21,3 +29,34 @@ def try_del(d, keys):
             del d[key]
         except KeyError:
             pass
+
+
+def _get_addresses_by_name(name):
```

Comment:
This could be decorated like this:

```python
@cache.Cache(_get_addresses_by_name)
def _addresses_by_name_cache(names):
    for name in names:
        # a set is used because this could contains duplicates
        addresses = {i[4][0] for i in socket.getaddrinfo(name, None)}

        # The value is stored in the cache before it is returned. The value is
        # translated to a tuple (instead of a list) to avoid being changed by
        # the caller. The tuple is sorted to make the result more predictable
        # as we do not have control over the order getaddrinfo and set sequence
        # yield values.
        yield name, tuple(sorted(addresses))
```